File: Algorithm/Programmers/Level 1/소수 찾기.py
def solution(n):
    answer = 0

    save = [True] * (n + 1)
    save[0:2] = False, False

    for i in range(2, n + 1):
        if save[i]:
            answer += 1
            for j in range(2 * i, n + 1, i):
                save[j] = False
    return answer

File: Algorithm/Programmers/Java/Level 1/소수_찾기.java
public class 소수_찾기 {
    public int solution(int n) {
        boolean[] b=new boolean[n+1];
        b[0]=true;
        b[1]=true;
        int answer = 0;
        for(int i=2; i<n+1; i++){
            if(b[i]==false){
                answer++;
                for(int j=i*2; j<n+1; j=j+i){
                    b[j]=true;
                }
            }
        }
        return answer;
    }
}
